guides/source/caching_with_rails.md

```diff
@@ -370,6 +370,18 @@ There are some common options that can be used by all cache implementations. The
 
 * `:race_condition_ttl` - This option is used in conjunction with the `:expires_in` option. It will prevent race conditions when cache entries expire by preventing multiple processes from simultaneously regenerating the same entry (also known as the dog pile effect). This option sets the number of seconds that an expired entry can be reused while a new value is being regenerated. It's a good practice to set this value if you use the `:expires_in` option.
 
+When using Puma or another threaded server it is recommended to use a connection pool to prevent thread contention. Connection pooling is only supported by the `MemCacheStore` and `RedisCacheStore`. To enable the connection pool first add the `connection_pool` gem to your Gemfile.
+
+```ruby
+gem 'connection_pool'
+```
+
+Next set either the `:pool_size` and/or `:pool_timeout` options.
```

Comment:
```suggestion
Next, set the `:pool_size` and/or `:pool_timeout` options:
```

I think this could use a code block example:

```ruby
config.cache_store = :mem_cache_store, "cache.example.com", pool_size: 5, pool_timeout: 5
```

Reply:
I've made the suggested changes and added the code example.